base/interactiveutil.jl

```diff
@@ -40,7 +40,7 @@ function edit(path::AbstractString, line::Integer=0)
     end
     background = true
     line_unsupported = false
-    if startswith(name, "emacs") || name == "gedit"
+    if startswith(name, "emacs") || name == "gedit" || startswith(name, "gvim")
```

Comment:
I have put it in line above because `gvim` is GUI based. Other `vim` variants run in console.
Therefore `gvim` should have `background = true` (as a separate window is opened when it is invoked) and other `vim`s  have `background = false`.